src/main/java/org/jboss/remoting3/EndpointBuilder.java

```diff
@@ -46,10 +47,15 @@ public final class EndpointBuilder {
     private List<ConnectionBuilder> connectionBuilders;
     private XnioWorker.Builder workerBuilder;
     //Default option map that sets heartbeat and read/write timeouts
-    private OptionMap defaultConnectionOptionMap = OptionMap.builder().set(RemotingOptions.HEARTBEAT_INTERVAL, RemotingOptions.DEFAULT_HEARTBEAT_INTERVAL)
+    private OptionMap defaultConnectionOptionMap = BUILTIN;
+
+    private static final OptionMap BUILTIN = OptionMap.builder().set(RemotingOptions.HEARTBEAT_INTERVAL, RemotingOptions.DEFAULT_HEARTBEAT_INTERVAL)
             .set(Options.READ_TIMEOUT, RemotingOptions.DEFAULT_HEARTBEAT_INTERVAL * 2)
             .set(Options.WRITE_TIMEOUT, RemotingOptions.DEFAULT_HEARTBEAT_INTERVAL * 2)
-            .set(Options.KEEP_ALIVE, Boolean.TRUE).getMap();
+            .set(Options.KEEP_ALIVE, Boolean.TRUE)
+            .set(RemotingOptions.MAX_INBOUND_CHANNELS, RemotingOptions.DEFAULT_MAX_INBOUND_CHANNELS)
```

Comment:
@gaol I think we don't need that. Could you please double check? Also, I'm not sure if we are going to correctly calculate READ_TIMEOUT and WRITE_TIMEOUT if (only) another HEARTBEAT_INTERVAL is provided

Reply:
> I think we don't need that. Could you please double check?

Ah, yes, I agree that we don't need `RemotingOptions.MAX_INBOUND_CHANNELS` and `RemotingOptions.MAX_OUTBOUND_CHANNELS` settings in the `BUILTIN`, the default value will be used if they are not defined. Would you please do the change ? :)

> Also, I'm not sure if we are going to correctly calculate READ_TIMEOUT and WRITE_TIMEOUT if (only) another HEARTBEAT_INTERVAL is provided

I don't think so, because we only provide the default settings, and users are free to set different values for their needs. unless there is a **fixed** relationship between the timeout and heartbeat.

Reply:
> > I think we don't need that. Could you please double check?
> 
> Ah, yes, I agree that we don't need `RemotingOptions.MAX_INBOUND_CHANNELS` and `RemotingOptions.MAX_OUTBOUND_CHANNELS` settings in the `BUILTIN`, the default value will be used if they are not defined. Would you please do the change ? :)
> 
fixed :)

> > Also, I'm not sure if we are going to correctly calculate READ_TIMEOUT and WRITE_TIMEOUT if (only) another HEARTBEAT_INTERVAL is provided
> 
> I don't think so, because we only provide the default settings, and users are free to set different values for their needs. unless there is a **fixed** relationship between the timeout and heartbeat.

Yeah true, agree